server/catgenome/src/main/java/com/epam/catgenome/manager/reference/MotifSearchManager.java

```diff
@@ -143,27 +145,59 @@ private MotifSearchResult searchChromosomeMotifs(final MotifSearchRequest reques
-        final Chromosome chr = loadChrById(motifSearchRequest.getReferenceId(), motifSearchRequest.getChromosomeId());
-        int start = motifSearchRequest.getStartPosition() == null
-                ? 0
-                : motifSearchRequest.getStartPosition();
-        int end = motifSearchRequest.getEndPosition() == null
-                ? chr.getSize()
-                : motifSearchRequest.getEndPosition();
-
+    private MotifSearchResult searchWholeGenomeMotifs(final MotifSearchRequest request) {
+        final int pageSize = request.getPageSize() == null ? DEFAULT_PAGE_SIZE : request.getPageSize();
+        int start = request.getStartPosition() == null ? 0 : request.getStartPosition();
+        Chromosome chromosome = loadChrById(request.getReferenceId(), request.getChromosomeId());
+        long chrId = chromosome.getId();
+        final int end = request.getEndPosition() == null ? chromosome.getSize() : request.getEndPosition();
+        if (end < chromosome.getSize()) {
+            return searchRegionMotifs(
+                    request.toBuilder()
+                            .pageSize(pageSize)
+                            .searchType(MotifSearchType.REGION)
+                            .chromosomeId(chrId)
+                            .startPosition(start)
+                            .endPosition(end)
+                            .build());
+        }
+        final List<Chromosome> chromosomes = getChromosomesOfGenome(request.getReferenceId());
+        final List<Motif> motifs = new ArrayList<>();
+        while (chromosome != null && pageSize - motifs.size() > 0) {
+            chrId = chromosome.getId();
+            motifs.addAll(searchChromosomeMotifs(
+                    request.toBuilder()
+                            .pageSize(pageSize - motifs.size())
+                            .searchType(MotifSearchType.CHROMOSOME)
+                            .chromosomeId(chrId)
+                            .startPosition(start)
+                            .endPosition(chromosome.getSize())
+                            .build())
+                    .getResult()
+                    .stream()
+                    .limit(pageSize - motifs.size())
+                    .collect(Collectors.toList()));
+            start = 0;
+            chromosome = getNextChromosome(chromosomes, chromosome);
+        }
         return MotifSearchResult.builder()
-                .result(
-                    fillMotifList(
-                        chr, start, end,
-                        motifSearchRequest.getPageSize(),
-                            motifSearchRequest.getStrand()))
-                .chromosomeId(chr.getId())
-                .pageSize(motifSearchRequest.getPageSize())
-                .position(end)
+                .result(motifs)
+                .pageSize(pageSize)
+                .chromosomeId(chrId)
+                .position(chromosome == null && pageSize - motifs.size() > 0 ? 0
```

Comment:
why you set position to 0 if `chromosome == null && pageSize - motifs.size() > 0`, as I understand in this case we should set position to null, it shows that there is no results left